File: Core/src/main/java/jokatu/game/player/PlayerFactory.java
package jokatu.game.player;

import jokatu.game.Game;
import org.jetbrains.annotations.NotNull;

/**
 * Creates players for users so they can interact with {@link Game}s.
 * @param <P> the type of {@link Player} to produce
 */
public interface PlayerFactory<P extends Player> {
	@NotNull
	P produce(@NotNull Game<?> game, @NotNull String username);
}

File: Core/src/main/kotlin/jokatu/game/player/PlayerFactory.kt
package jokatu.game.player

import jokatu.game.Game

/**
 * Creates players for users so they can interact with [Game]s.
 * @param <P> the type of [Player] to produce
 */
interface PlayerFactory<P : Player> {
	fun produce(game: Game<*>, username: String): P
}
